internal/codecs/base_scalar.go

```diff
@@ -524,21 +522,20 @@ func (c *Duration) Type() reflect.Type {
 }
 
 // Decode a duration.
-func (c *Duration) Decode(buf *buff.Buff, out unsafe.Pointer) {
-	buf.PopUint32() // data length
-	microseconds := int64(buf.PopUint64())
-	buf.PopUint32() // reserved
-	buf.PopUint32() // reserved
+func (c *Duration) Decode(r *buff.Reader, out unsafe.Pointer) {
+	r.Discard(4) // data length
```

Comment:
Why are you discarding data length in all decoders? You should actually read it and enforce it in every codec. In Python we have the `FRBuffer` helper to do that very efficiently (and `ReadBuffer` in JS), worth taking a look.

Reply:
They are discarded because it didn't matter on day 1. It makes sense to check theme though in real world scenarios. Created an issue for this https://github.com/edgedb/edgedb-go/issues/44